app/services/EssttpBarsService.scala

```diff
@@ -42,14 +49,30 @@ class EssttpBarsService @Inject() (barsService: BarsService, auditService: Audit
   )(implicit requestHeader: RequestHeader): Future[Either[BarsError, VerifyResponse]] = {
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(requestHeader)
 
+    val taxId = journey match {
+      case account: Stages.ChosenTypeOfBankAccount     => account.taxId
```

Comment:
can you get away with `Journey.AfterChosenTypeOfBankAccount` or similar here maybe? to avoid duplicates

Reply:
not sure, I can check though. IntelliJ automatically generated those case statements